xmake/modules/utils/binary/deplibs.lua

```diff
@@ -81,7 +81,7 @@ function _get_all_depends_by_objdump(binaryfile, opt)
                 else
                     if line:startswith("NEEDED") then
                         local filename = line:split("%s+")[2]
-                        if filename and filename:endswith(".so") then
+                        if filename and filename:endswith(".so") or filename:match(".-%.so%.%d+") then
```

Comment:
I think we should use `filename:find("%.so%.%d+$")` instead of `filename:match(".-%.so%.%d+")`